src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java

```diff
@@ -553,6 +574,21 @@ public boolean cancel() {
       return !future.cancel(true);
     }
 
+    @Override
+    public void close() {
+      // Call register() to ensure arriveAndDeregister() will terminate the
+      // downloadPhaser if the download has not started yet.
+      if (downloadPhaser.register() < 0) {
+        // The download completed normally or has already been cancelled and the thread has
+        // terminated in response to the interrupt.
+        return;
+      }
+      try (SilentCloseable c =
+          Profiler.instance().profile("Cancelling download " + outputPath)) {
+        downloadPhaser.awaitAdvance(downloadPhaser.arriveAndDeregister());
```

Comment:
```suggestion
        downloadPhaser.arriveAndAwaitAdvance();
```
should suffice as we don't care about deregistering (there won't be another phase).

Reply:
The code was written to use Phaser termination as the signalling method. I've now changed to to use `phase=0` as signalling method according to the suggestion. The JSON profiles look very nice in the small test case.